packages/modules/create-imodel/src/index.ts

```diff
@@ -2,23 +2,31 @@
  * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
-export { CreateIModel } from "./components/create-imodel/CreateIModel";
+import { ButtonBar as ButtonBarDefault } from "./components/button-bar";
+import { CreateIModel as CreateIModelDefault } from "./components/create-imodel/CreateIModel";
+import { IModelDescription as IModelDescriptionDefault } from "./components/imodel-description/IModelDescription";
+import { IModelName as IModelNameDefault } from "./components/imodel-name/IModelName";
+import { UpdateIModel as UpdateIModelDefault } from "./components/update-imodel/UpdateIModel";
+import { UploadImage as UploadImageDefault } from "./components/upload-image/UploadImage";
+import { withThemeProvider } from "./utils/WithThemeProvider";
+
+export const CreateIModel = withThemeProvider(CreateIModelDefault);
 export type { CreateIModelProps } from "./components/create-imodel/CreateIModel";
 
-export { UpdateIModel } from "./components/update-imodel/UpdateIModel";
+export const UpdateIModel = withThemeProvider(UpdateIModelDefault);
 export type { UpdateIModelProps } from "./components/update-imodel/UpdateIModel";
 
 export { IModelFull } from "./types";
 export type { BaseIModel, iModelExtent, ExtentPoint } from "./types";
 
-export { IModelName } from "./components/imodel-name/IModelName";
+export const IModelName = withThemeProvider(IModelNameDefault);
```

Comment:
wait... why...
YOu dont need to wrap all of the inner sub components with theme provider.. Just wrap the root one! 
Those inner components cannot be used outside of our provided context/root component..
This looks messy and knowing you applied custom style, might mess up something

Reply:
Reverted changes, please have a look